src/mixin/java/io/github/brassmc/brassloader/mixin/BootstrapMixin.java

```diff
@@ -0,0 +1,48 @@
+package io.github.brassmc.brassloader.mixin;
+
+import io.github.brassmc.brassloader.boot.discovery.InvalidEntrypointException;
+import io.github.brassmc.brassloader.boot.discovery.ModDiscovery;
+import io.github.brassmc.brassloader.boot.mods.ModContainer;
+import net.minecraft.server.Bootstrap;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+
+@Mixin(Bootstrap.class)
+public class BootstrapMixin {
+    @Inject(
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/core/Registry;freezeBuiltins()V"
+            ),
+            method = "bootStrap"
+    )
+    private static void init(CallbackInfo callback) {
+        System.out.println("Found mods: [" + String.join(", ", ModDiscovery.MODS.stream().map(ModContainer::modid).toArray(String[]::new)) + "]");
+        ModDiscovery.MODS.forEach(modContainer -> {
+            String entrypoint = modContainer.entrypoint();
+            try {
+                Class<?> mainClass = Class.forName(entrypoint);
+                Constructor<?> constructor = mainClass.getConstructor();
+                constructor.setAccessible(true);
+                constructor.newInstance();
+            } catch (ClassNotFoundException exception) {
+                throw new InvalidEntrypointException("Entrypoint in mod [" + modContainer.modid() + "] points to an invalid class!");
+            } catch (InvocationTargetException exception) {
+                throw new InvalidEntrypointException("Entrypoint in mod [" + modContainer.modid() + "] threw an exception!");
+            } catch (InstantiationException exception) {
+                throw new InvalidEntrypointException("Entrypoint in mod [" + modContainer.modid() + "] is abstract!");
+            } catch (IllegalAccessException exception) {
+                throw new InvalidEntrypointException("Entrypoint in mod [" + modContainer.modid() + "] is not accessible!");
+            } catch (NoSuchMethodException exception) {
+                throw new InvalidEntrypointException("Entrypoint in mod [" + modContainer.modid() + "] does not contain a no-args constructor!");
+            }
```

Comment:
Collapse this block into a `catch (Exception)`, the Java exceptions messages are detailed enough.

Reply:
I prefer to handle this myself rather than the less descriptive default implementations.

Reply:
In that case, please add the original exception as a parent of the new exception.